### ISSUE TITLE
CD compiler does not report unresolved imported types

### ISSUE THREAD

**Issue by schmalzing** [MontiCore/cd4analysis#118]
## Issue description

When compiling a MontiCore class diagram, an invalid import is accepted without an error. The problem is only detected later when the generated Java code is compiled.

## Reproduction

`montiarc.generator.meta.Java.cd`:

```cd
classdiagram Java {
  public abstract class Class { }
}
```

`montiarc.generator.meta.MA2Jsim.cd`:

```cd
import montiarc.generator.Java.Class;

classdiagram MA2Jsim {
  public class MCConcept extends Class { }
}
```

The import in `MA2Jsim.cd` is invalid. The declared type is actually:

```text
montiarc.generator.meta.Java.Class
```

## Actual behavior

The CD compilation succeeds and generates Java code containing:

```java
import montiarc.generator.Java.Class;
```

Java compilation then fails, for example:

```text
ERROR in ...\MA2Jsim\Transition.java (at line 21)
import montiarc.generator.Java.Class;
       ^^^^^^^^^^^^^^^^^^^^^^^
The import montiarc.generator.Java cannot be resolved
```

## Expected behavior

The CD compiler should report an error during CD processing because `montiarc.generator.Java.Class` cannot be resolved.

The correct import is:

```cd
import montiarc.generator.meta.Java.Class;
```

## Impact

Invalid cross-CD dependencies remain undetected until the Java-generation/Java-compilation phase. This delays feedback and makes the error appear as a generated-code or Java classpath problem rather than an invalid CD import.

**Comment by luepges** [comment]
-> a CD4Code CoCo (that is added by the generator) should catch this

**Comment by luepges** [comment]
Problem:

````
classdiagram A {
 class AC {}
}
---
import A.AC;
classdiagram B {
  class BC extends AC {}
}
````

In the current setup the diagrams are built the following way:
* `java -jar ... A.cd`
* `java -jar ... B.cd`

Thus, no A-symbols exist for classdiagram B.

Potential Solution:
* `java -jar ... A.cd B.cd` and build all classdiagrams sequentially from the same tool (parse both --> ... -> generate both)?
Nvm: This is already done (just not in some test cases)
